src/renderer/components/version-select.tsx

```diff
@@ -58,15 +58,32 @@ export function getItemIcon({ state }: RunnableVersion) {
  * Helper method: Returns the <Select /> predicate for an Electron
  * version.
  *
+ * Sorts by index of the chosen query.
+ * For example, if we take the following versions:
+ * [3.0.0, 14.3.0, 13.2.0, 12.0.0-nightly.20210301, 12.0.0-beta.3]
+ * and a search query of '3', this method would sort them into:
+ * [3.0.0, 13.2.0, 14.3.0, 12.0.0-beta.3, 12.0.0-nightly.20210301]
+ *
  * @param {string} query
- * @param {RunnableVersion} { version }
+ * @param {RunnableVersion[]} versions
  * @returns
  */
-export const filterItem: ItemPredicate<RunnableVersion> = (
+export const filterItems: ItemListPredicate<RunnableVersion> = (
   query,
-  { version },
+  versions,
 ) => {
-  return version.toLowerCase().includes(query.toLowerCase());
+  if (query === '') return versions;
+
+  const q = query.toLowerCase();
+
+  return versions
+    .filter(({ version }) => version.toLowerCase().includes(q))
+    .sort((a, b) => {
+      if (a.version.indexOf(q) > b.version.indexOf(q)) return 1;
+      if (b.version.indexOf(q) > a.version.indexOf(q)) return -1;
+
+      return 0;
+    });
```

Comment:
Since you commented on perf -- we can bring the number of searches down from O(n^2) to O(n) with a temporary, and that lets us perform the sort on a lowercase set as well

```suggestion
  return versions
    .map((version: RunnableVersion) => ({
      index: version.version.toLowerCase().indexOf(q),
      version,
    }))
    .filter((item) => item.index !== -1)
    .sort((a, b) => a.index - b.index)
    .map((item) => item.version);
```

Reply:
ooh i didn't consider the temporary! good shout, ty!